File: frontend/src/utils/stateLanguageMapping.js
// Default State-Language Mapping for India
// Maps each state to its primary and secondary languages (array format)

export const DEFAULT_STATE_LANGUAGE_MAPPING = {
  'all': ['Hindi'],  // National/Bollywood releases
  'ap': ['Telugu'],  // Andhra Pradesh
  'ar': ['English', 'Hindi'], // Arunachal Pradesh
  'as': ['Assamese', 'Bengali'], // Assam
  'br': ['Hindi', 'Bhojpuri'],   // Bihar
  'cg': ['Hindi'],   // Chhattisgarh
  'dl': ['Hindi', 'Punjabi', 'Urdu'],   // Delhi
  'ga': ['Konkani', 'Marathi'], // Goa
  'gj': ['Gujarati', 'Hindi'], // Gujarat
  'hr': ['Hindi'],   // Haryana
  'hp': ['Hindi'],   // Himachal Pradesh
  'jk': ['Urdu', 'Hindi', 'Kashmiri'],    // Jammu and Kashmir
  'jh': ['Hindi'],   // Jharkhand
  'ka': ['Kannada', 'Telugu'], // Karnataka
  'kl': ['Malayalam'], // Kerala
  'ld': ['English', 'Hindi'], // Ladakh
  'mp': ['Hindi'],   // Madhya Pradesh
  'mh': ['Marathi', 'Hindi'], // Maharashtra
  'mn': ['Meitei', 'English'],  // Manipur
  'ml': ['English', 'Khasi'], // Meghalaya
  'mz': ['Mizo', 'English'],    // Mizoram
  'nl': ['English'],            // Nagaland
  'or': ['Odia'],    // Odisha
  'pb': ['Punjabi', 'Hindi'], // Punjab
  'rj': ['Hindi'],   // Rajasthan
  'sk': ['Nepali', 'English'],  // Sikkim
  'tn': ['Tamil'],   // Tamil Nadu
  'ts': ['Telugu'],  // Telangana
  'tr': ['Bengali', 'Kokborok'], // Tripura
  'up': ['Hindi', 'Urdu'],   // Uttar Pradesh
  'uk': ['Hindi'],   // Uttarakhand
  'wb': ['Bengali', 'Hindi']  // West Bengal
};

// Available languages for selection
export const AVAILABLE_LANGUAGES = [
  'Hindi',
  'Telugu',
  'Tamil',
  'Kannada',
  'Malayalam',
  'Bengali',
  'Marathi',
  'Gujarati',
  'Punjabi',
  'Odia',
  'Assamese',
  'Urdu',
  'Bhojpuri',
  'Konkani',
  'Nepali',
  'Meitei',
  'Mizo',
  'Kashmiri',
  'Khasi',
  'Kokborok',
  'English'
];

// Get languages for a state code (returns array)
export const getLanguagesForState = (stateCode) => {
  const languages = DEFAULT_STATE_LANGUAGE_MAPPING[stateCode];
  if (!languages) return ['Hindi'];
  return Array.isArray(languages) ? languages : [languages];
};

// Get primary language for a state (first in array)
export const getLanguageForState = (stateCode) => {
  const languages = getLanguagesForState(stateCode);
  return languages[0] || 'Hindi';
};

// Get all states with their languages
export const getAllStateMappings = () => {
  return DEFAULT_STATE_LANGUAGE_MAPPING;
};

File: backend/state_language_mapping.py
# Default State-Language Mapping for India

DEFAULT_STATE_LANGUAGE_MAPPING = {
    'all': 'Hindi',  # National/Bollywood releases
    'ap': 'Telugu',  # Andhra Pradesh
    'ar': 'English', # Arunachal Pradesh
    'as': 'Assamese', # Assam
    'br': 'Hindi',   # Bihar
    'cg': 'Hindi',   # Chhattisgarh
    'dl': 'Hindi',   # Delhi
    'ga': 'Konkani', # Goa
    'gj': 'Gujarati', # Gujarat
    'hr': 'Hindi',   # Haryana
    'hp': 'Hindi',   # Himachal Pradesh
    'jk': 'Urdu',    # Jammu and Kashmir
    'jh': 'Hindi',   # Jharkhand
    'ka': 'Kannada', # Karnataka
    'kl': 'Malayalam', # Kerala
    'ld': 'English', # Ladakh
    'mp': 'Hindi',   # Madhya Pradesh
    'mh': 'Marathi', # Maharashtra
    'mn': 'Meitei',  # Manipur
    'ml': 'English', # Meghalaya
    'mz': 'Mizo',    # Mizoram
    'nl': 'English', # Nagaland
    'or': 'Odia',    # Odisha
    'pb': 'Punjabi', # Punjab
    'rj': 'Hindi',   # Rajasthan
    'sk': 'Nepali',  # Sikkim
    'tn': 'Tamil',   # Tamil Nadu
    'ts': 'Telugu',  # Telangana
    'tr': 'Bengali', # Tripura
    'up': 'Hindi',   # Uttar Pradesh
    'uk': 'Hindi',   # Uttarakhand
    'wb': 'Bengali'  # West Bengal
}

def get_language_for_state(state_code):
    """Get the primary language for a state code"""
    return DEFAULT_STATE_LANGUAGE_MAPPING.get(state_code, 'Hindi')

def get_languages_for_states(state_codes):
    """Get list of languages for multiple state codes"""
    languages = []
    for state_code in state_codes:
        lang = get_language_for_state(state_code)
        if lang and lang not in languages:
            languages.append(lang)
    return languages
